nessie/sql_templates/create_ycbm_schema.template.sql

```diff
@@ -51,3 +51,72 @@ CREATE EXTERNAL TABLE {redshift_schema_ycbm}.bookings(
 )
 ROW FORMAT SERDE 'org.openx.data.jsonserde.JsonSerDe'
 LOCATION '{loch_s3_ycbm_data_path}/archive';
+
+--------------------------------------------------------------------
+-- Internal schema
+--------------------------------------------------------------------
+
+DROP SCHEMA IF EXISTS {redshift_schema_ycbm_internal} CASCADE;
+CREATE SCHEMA {redshift_schema_ycbm_internal};
+GRANT USAGE ON SCHEMA {redshift_schema_ycbm_internal} TO GROUP {redshift_app_boa_user}_group;
+ALTER default PRIVILEGES IN SCHEMA {redshift_schema_ycbm_internal} GRANT SELECT ON TABLES TO GROUP {redshift_app_boa_user}_group;
+GRANT USAGE ON SCHEMA {redshift_schema_ycbm_internal} TO GROUP {redshift_dblink_group};
+ALTER DEFAULT PRIVILEGES IN SCHEMA {redshift_schema_ycbm_internal} GRANT SELECT ON TABLES TO GROUP {redshift_dblink_group};
+
+--------------------------------------------------------------------
+-- Internal tables
+--------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION {redshift_schema_ycbm_internal}.to_utc_iso_string(date_string VARCHAR)
+RETURNS VARCHAR
+STABLE
+AS $$
+  from datetime import datetime
+  import pytz
+
+  d = datetime.strptime(date_string, '%Y-%m-%dT%H:%M:%S')
+  d = pytz.timezone('America/Los_Angeles').localize(d)
+  return d.astimezone(pytz.utc).isoformat()
```

Comment:
Nifty!

Reply:
I stole it from someone else's code (I think @lyttam) but nifty it is!